lib/App.js

```diff
@@ -235,7 +235,7 @@ App.prototype.component = function(name, constructor, isDependency) {
 
   // Calling app.component() overrides existing views or components. Prevent
   // dependencies from doing this without warning
-  if (isDependency && currentView) {
+  if (isDependency && currentView && !serializedViews) {
```

Comment:
This change causes the error never be thrown, since `serializedViews` is an empty module on the server and thus is truthy:

```
$ node
> require('./lib/_views')
{}
> !!require('./lib/_views')
true
```

In a client using serialized views, [the variable's a function](https://github.com/derbyjs/derby/blob/v0.10.24/lib/App.js#L56), so `var usingSerializedViews = typeof serializedViews === 'function';` at module level could work.